### PR TITLE
Another way to unify a declaration location for NetworkModule

### PR DESCRIPTION
**Pull request by jmatsu**
## Issue
- ref https://github.com/DroidKaigi/conference-app-2018/pull/213

## Overview (Required)
- Declare ~flavored~ build type specific modules for each ~flavor~ build types
- Inject such items via a custom Qualitifer

### CODE REVIEW

File: app/src/release/java/io/github/droidkaigi/confsched2018/di/BuildTypeBasedNetworkModule.kt [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+import dagger.multibindings.IntoSet
+import okhttp3.Interceptor
+import javax.inject.Singleton
+
+@Module
+internal class BuildTypeBasedNetworkModule {
+
+    @NetworkLogger @Singleton @Provides @IntoSet
+    fun provideNetworkLogger(): Interceptor = HttpLoggingInterceptor().apply {
+        level = HttpLoggingInterceptor.Level.NONE
```
Comment (takahirom) [inline app/src/release/java/io/github/droidkaigi/confsched2018/di/BuildTypeBasedNetworkModule.kt]: We selected BASIC for release. What do you think? 🐇 
https://github.com/DroidKaigi/conference-app-2018/pull/50#discussion_r160300524

File: app/src/main/java/io/github/droidkaigi/confsched2018/di/NetworkModule.kt [hunk 2]
```diff
@@ -11,22 +10,26 @@ import io.github.droidkaigi.confsched2018.data.api.GithubApi
...
+)
+internal object NetworkModule {
 
     @Singleton @Provides @JvmStatic
-    fun provideOkHttpClient(): OkHttpClient = OkHttpClient.Builder()
-            .addNetworkInterceptor(HttpLoggingInterceptor()
-                    .setLevel(HttpLoggingInterceptor.Level.BODY))
-            .addNetworkInterceptor(StethoInterceptor())
-            .build()
+    fun provideOkHttpClient(@NetworkLogger loggingINterceptors: Set<Interceptor>): OkHttpClient =
```
Comment (takahirom) [inline app/src/main/java/io/github/droidkaigi/confsched2018/di/NetworkModule.kt]: loggingINterceptors -> loggingInterceptors ℹ️

Comment (jmatsu) [reply app/src/main/java/io/github/droidkaigi/confsched2018/di/NetworkModule.kt]: 😂

**Comment by jmatsu** [comment]
Thanks @takahirom ! Addressed 👍

**Comment by jmatsu** [comment]
mmm, sorry, my implementation looks wrong. let me check.

**Comment by jmatsu** [comment]
I forgot that this is Kotlin project 🤣 Suppressed wildcard!

File: app/src/main/java/io/github/droidkaigi/confsched2018/di/NetworkModule.kt [hunk 3]
```diff
@@ -11,22 +10,28 @@ import io.github.droidkaigi.confsched2018.data.api.GithubApi
...
+)
+internal object NetworkModule {
 
     @Singleton @Provides @JvmStatic
-    fun provideOkHttpClient(): OkHttpClient = OkHttpClient.Builder()
-            .addNetworkInterceptor(HttpLoggingInterceptor()
-                    .setLevel(HttpLoggingInterceptor.Level.BODY))
-            .addNetworkInterceptor(StethoInterceptor())
-            .build()
+    fun provideOkHttpClient(@NetworkLogger loggingInterceptors: Set<@JvmSuppressWildcards
```
Comment (takahirom) [inline app/src/main/java/io/github/droidkaigi/confsched2018/di/NetworkModule.kt]: 📝

**Comment by takahirom** [comment]
Looks good!! Thanks! 🗡

**Pull request merged**